src/components/MessageButton.ts

```diff
@@ -0,0 +1,281 @@
+import type { APIPartialEmoji, Snowflake } from 'discord-api-types/v8';
+import ow from 'ow';
+import { APIBaseComponent, ComponentType } from './BaseComponent';
+
+/**
+ * Builds a button-type message component
+ */
+export class MessageButton {
+	/**
+	 * The text to be displayed on this button
+	 * @type {?string}
+	 */
+	public customID?: string;
```

Comment:
```suggestion
	public customId?: string;
```

Reply:
We'll not use this in v13 anyway, right? cc @discordjs/the-big-3 

Otherwise we'll have to change this once we go to the rewrite, which is a bit iffy I feel.

Reply:
We can always break this for v13, depends on when you want to release it.

Reply:
I opened https://github.com/discordjs/discord.js/pull/6036 a few hours ago, so depending on whether we go for `ID` or `Id`, depends on whether or not we land that PR for v13.

Reply:
Yup, merged, we can move forward with that change.